src/components/PopupWithForm.js

```diff
@@ -24,7 +26,13 @@ export default class PopupWithForm extends Popup {
     });
   }
 
+  setButtonText(text) {
+    this._submitButton.textContent = text;
+  }
+
   close() {
     super.close();
+    this._form.reset(); // ✅ Optional: Reset fields on close
+    this.setButtonText(this._defaultButtonText); // ✅ Restore default button text
```

Comment:
You need to **only** return the **default** button text in the `finally` block, so that the button could get the text regardless of whether the request succeeds or fails.